deploy/ray-operator/api/v1/raycluster_types.go

```diff
@@ -0,0 +1,105 @@
+package v1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// RayClusterSpec defines the desired state of RayCluster
+type RayClusterSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+	ClusterName     string          `json:"clusterName"`
+	Images          RayClusterImage `json:"images"`
+	ImagePullPolicy string          `json:"imagePullPolicy,omitempty"`
+	Extensions      []Extension     `json:"extensions,omitempty"`
+}
+
+// RayClusterStatus defines the observed state of RayCluster
+type RayClusterStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// +kubebuilder:object:root=true
+
+// RayCluster is the Schema for the RayClusters API
+type RayCluster struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   RayClusterSpec   `json:"spec,omitempty"`
+	Status RayClusterStatus `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// RayClusterList contains a list of RayCluster
+type RayClusterList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []RayCluster `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&RayCluster{}, &RayClusterList{})
+}
+
+type RayClusterImage struct {
+	DefaultImage string `json:"defaultImage,omitempty"`
+}
+
+type Extension struct {
+	// the size of this pod group
+	Size int32 `json:"size"`
+
+	// worker/head
+	Type string `json:"type,omitempty"`
+
+	// pod image
+	Image string `json:"image,omitempty"`
+
+	// logic groupName for worker in same group, can be functional or in size meaning
```

Comment:
```suggestion
	// logical groupName for worker in same group, can be functional or in size meaning
```
I don't know what this comment means. What is a worker group?

Reply:
Group, like the name tells, will be used to divide the nodes in a ray cluster into groups. Nodes in different groups could have 1. different roles, e.g. header vs. worker. 2. different configurations which can be used to setup a ray cluster with heterogeneous nodes.